connection/src/test/java/com/netcracker/ConnectionTest.java

```diff
@@ -0,0 +1,99 @@
+package com.netcracker;
+
+import com.netcracker.domain.Message;
+import org.junit.Test;
+
+import java.io.*;
+import java.net.Socket;
+import java.net.SocketAddress;
+
+import static com.netcracker.domain.MessageType.TEXT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ConnectionTest {
+    private static Connection connection;
+    private static ByteArrayOutputStream innerOutputStream;
+    private static int originalNumberOfBytes;
+    private static final SocketAddress address = mock(SocketAddress.class);
+    private static final Message messageToReceive = new Message(TEXT, "MessageToReceive");
+    private static final Message messageToSend = new Message(TEXT, "MessageToSend");
+    private static final Message testMessage = new Message(TEXT, "InOrderNotToGetEOFException");
+
+    private void setUpConnectionObjectWithByteStreams() throws Exception {
+        ByteArrayOutputStream outerOutputStream = new ByteArrayOutputStream();
+        try (ObjectOutputStream outerOOS = new ObjectOutputStream(outerOutputStream)) {
+            outerOOS.writeObject(messageToReceive);
+        }
+
+        innerOutputStream = new ByteArrayOutputStream();
+        try (ObjectOutputStream oos = new ObjectOutputStream(innerOutputStream)) {
+            oos.writeObject(testMessage);
+        }
+        originalNumberOfBytes = innerOutputStream.toByteArray().length;
+
+        ByteArrayInputStream innerInputStream = new ByteArrayInputStream(outerOutputStream.toByteArray());
```

Comment:
Yeah, I did indeed ;)
The thing is, these two requirements that I gave you are not (at least if you look at them in the proper way) mutually exclusive.
The first one says: "You should initialize variable at the closest point possible to its intended usage." - this one is true with the minor correction (which I describe below).

You looked at it and "obeyed without giving it much thought" and as the result didn't do it all the way through. If you look at your code closely, when you write `ByteArrayInputStream innerInputStream = new ByteArrayInputStream(outerOutputStream.toByteArray());` you use `outerOutputStream` which is initialized/created "not at the closest point possible". That means if your object depends on other objects, these dependencies should also stick close to your object.